pkg/solana/client/rpc_client.go

```diff
@@ -0,0 +1,312 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"golang.org/x/sync/singleflight"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	mn "github.com/smartcontractkit/chainlink-solana/pkg/solana/client/multinode"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/monitor"
+)
+
+var _ ReaderWriter = (*RPCClient)(nil)
+
+type Head struct {
+	rpc.GetBlockResult
+}
+
+func (h *Head) BlockNumber() int64 {
+	if h.BlockHeight == nil {
+		return 0
+	}
+	return int64(*h.BlockHeight)
+}
+
+func (h *Head) BlockDifficulty() *big.Int {
+	return nil
+}
+
+func (h *Head) IsValid() bool {
+	return true
+}
+
+type RPCClient struct {
+	url             string
+	rpc             *rpc.Client
+	skipPreflight   bool // to enable or disable preflight checks
+	commitment      rpc.CommitmentType
+	maxRetries      *uint
+	txTimeout       time.Duration
+	contextDuration time.Duration
+	log             logger.Logger
+
+	// provides a duplicate function call suppression mechanism
+	requestGroup *singleflight.Group
+}
+
+// TODO: BCI-4061: Implement RPC Client for MultiNode
+
+func (c *RPCClient) Dial(ctx context.Context) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) SubscribeToHeads(ctx context.Context) (<-chan *Head, mn.Subscription, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) SubscribeToFinalizedHeads(ctx context.Context) (<-chan *Head, mn.Subscription, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) Ping(ctx context.Context) error {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) IsSyncing(ctx context.Context) (bool, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) UnsubscribeAllExcept(subs ...mn.Subscription) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) Close() {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (c *RPCClient) GetInterceptedChainInfo() (latest, highestUserObservations mn.ChainInfo) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func NewRPCClient(endpoint string, cfg config.Config, requestTimeout time.Duration, log logger.Logger) (*RPCClient, error) {
```

Comment:
isn't this just a copy paste of the `client.go` code with additional methods for multinode?

why not do something like:

```go
type MultiNodeClient struct {
	Client
}

<add multinode specific methods>
```

Reply:
We can actually just extend the existing client instead of needing this. Updated :)